### PR TITLE
add prefect to qhub

### PR DESCRIPTION
**Pull request by tylerpotts**
This adds prefect to a qhub deployment when the flag `prefect: True` is set

### CODE REVIEW

File: qhub/template/{{ cookiecutter.repo_directory }}/infrastructure/prefect.tf [hunk 1]
```diff
@@ -0,0 +1,19 @@
++resource "helm_release" "prefect" {
++  name      = "prefect"
++  namespace = var.environment
++  chart     = "../charts/prefect"
```
Comment (aktech) [inline qhub/template/{{ cookiecutter.repo_directory }}/infrastructure/prefect.tf:4]: This does not exists here, we need to create a prefect module in [terraform modules](https://github.com/Quansight/qhub-terraform-modules) and call that module.

Comment (aktech) [reply qhub/template/{{ cookiecutter.repo_directory }}/infrastructure/prefect.tf:4]: Also, would need to add some usage documentation.

Comment (tylerpotts) [reply qhub/template/{{ cookiecutter.repo_directory }}/infrastructure/prefect.tf:4]: https://github.com/Quansight/qhub-datum-gcp-prod_old_2020-11-12/blob/master/infrastructure/prefect.tf

Comment (tylerpotts) [reply qhub/template/{{ cookiecutter.repo_directory }}/infrastructure/prefect.tf:4]: Example in infrastructure/prefect.tf for old client deployment

**Comment by aktech** [comment]
- [ ] TODO for myself: Check the the Github actions file have prefect token as variable

**Comment by tylerpotts** [comment]
LGTM

**Pull request merged**